controllers/topology/util.go

```diff
@@ -68,3 +69,8 @@ func refToUnstructured(ref *corev1.ObjectReference) *unstructured.Unstructured {
 	uns.SetName(ref.Name)
 	return uns
 }
+
+// objError creates an error based on the given err, msg and object.
+func objError(err error, msg string, object client.Object) error {
```

Comment:
Is the idea to make this a pattern going forward? fmt.Errorf is used in a bunch of places across the code, and it's a well known function. This helper makes things easier to read in some ways, but if it's not an accepted pattern across the codebase it could make things harder to follow.

Reply:
The goal was to avoid repetition in the way we construct our errors in the topology controller. And also to make them more consistent. There were already a few cases where we built them differently, in some cases they were even wrong (we had `<gvk>/kind`)

I'll wait for further feedback. If we don't want it or someone has a better idea how to do it I'm happy to adjust the PR accordingly.